tasks/main.yml

```
@@ -23,3 +39,17 @@
       # See: https://github.com/ansible/ansible-lint/issues/1894
       command: "{{ nix_script_dir }}/install_nix.sh"  # noqa no-changed-when
   when: nix_directory is changed  # noqa no-handler
+
+- name: Set up nix flakes
+  block:
+    - name: Install nix flakes package
+      become: true
+      become_flags: "--login"
+      become_user: "{{ nix_user }}"
+      shell: "nix-env -iA nixpkgs.nixFlakes"
+
+    - name: Enable nix flakes
+      become: true
+      become_user: "{{ nix_user }}"
+      shell: "echo 'experimental-features = nix-command flakes' >>  \"{{ nix_config_dir.path }}/nix.conf\""
```

Why not use `lineinfile`?

```suggestion
      shell: "echo 'experimental-features = nix-command flakes' >>  \"{{ nix_config_dir.path }}/nix.conf\""
      lineinfile:
        path: "{{ nix_config_dir.path }}/nix.conf"
        regexp: "^experimental-features"
        line: "experimental-features = 'nix-command flakes'"
        owner: "{{ nix_user }}"
        create: true
        mode: "0644"
        state: present
```

Note that this would also make the `become_user` property unnecessary.

The problem with my suggested approach is that if there is other stuff already defined in `experimental-features`, it would be overwritten and lost. It would be possible to preserve this with a capture expression in the `regexp`, but I'm not sure what the `nix.conf` format looks like with respect to multiple arguments. I did a bit of quick googling and couldn't find any suitable examples. Do you know how one would specify multiply things here?

---

They're space-separated I think ([`nix-command`](https://nixos.wiki/wiki/Nix_command) is one config option)

---

Ok, that's good. However, I realized that it's a bit tricky to do `regexp: "^experimental-features = '(.*)'"` and then to use that match in the `line`, because then running this task multiple times would result in `experimental-features = 'nix-command flakes nix-command flakes'`.

However, the current approach in this PR is also not great because if the nix config file already existed with `experimental-features` in it, then there would be two lines (with possible differing content).

I think maybe the best approach here is to just take over the nix config file. :smile: In other words, maybe instead of `lineinfile` we could use `blockinfile`, which inserts the markers that make it clear that it's an Ansible-managed file and should not be edited manually. I've revised my suggestion above, please let me know what you think.